src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,283 @@
+                var securityIdForTheUser = nativeWindowsHelper.GetSecurityId(domainName, userName);
+                AutoLogonRegistryManager regHelperForDiffUser = new AutoLogonRegistryManager(regManager, securityIdForTheUser);
+                regHelperForDiffUser.RevertOriginalRegistrySettings();
+            }            
+        }
+
+        public bool IsAutoLogonConfigured()
+        {
+            //ToDo: Different user scenario
+            
+            //find out the path for startup process if it is same as current agent location, yes it was configured
+            var regHelper = new AutoLogonRegistryManager(HostContext.GetService<IWindowsRegistryManager>(), null);
+            var startupCommand = regHelper.GetStartupProcessCommand();
+
+            if(string.IsNullOrEmpty(startupCommand))
+            {
+                return false;
+            }
+
+            var expectedStartupProcessDir = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin));
+            return Path.GetDirectoryName(startupCommand).Equals(expectedStartupProcessDir, StringComparison.CurrentCultureIgnoreCase);
+        }
+
+        private void UpdateRegistriesForAutoLogon(AutoLogonRegistryManager regHelper, string userName, string domainName, string logonPassword)
+        {
+            regHelper.UpdateStandardRegistrySettings();
+
+            //auto logon
+            ConfigureAutoLogon(regHelper, userName, domainName, logonPassword);
+
+            //startup process
+            var startupProcessPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), "agent.listener.exe");
+            var startupCommand = string.Format($@"{startupProcessPath} run");
+            Trace.Verbose($"Setting startup command as {startupCommand}");
+
+            regHelper.SetStartupProcessCommand(startupCommand);
+        }
+
+        private void ConfigureAutoLogon(AutoLogonRegistryManager regHelper, string userName, string domainName, string logonPassword)
+        {
+            //find out if the autologon was already enabled, show warning in that case
+            ShowAutoLogonWarningIfAlreadyEnabled(regHelper, userName);
+
+            var windowsHelper = HostContext.GetService<INativeWindowsServiceHelper>();
+            windowsHelper.SetAutoLogonPassword(logonPassword);
+
+            regHelper.UpdateAutoLogonSettings(userName, domainName);
+        }
+
+        private void ShowAutoLogonWarningIfAlreadyEnabled(AutoLogonRegistryManager regHelper, string userName)
+        {
+            regHelper.FetchAutoLogonUserDetails(out string autoLogonUserName, out string domainName);
+            if(autoLogonUserName != null && !userName.Equals(autoLogonUserName, StringComparison.CurrentCultureIgnoreCase))
+            {
+                _terminal.WriteLine(string.Format(StringUtil.Loc("AutoLogonAlreadyEnabledWarning"), userName));
```

Comment:
don't use string.format, pass the format args to the loc function. it will pass the correct culture and perform the string format in a safe way.